components/camel-dynamic-router/src/main/java/org/apache/camel/component/dynamicrouter/DynamicRouterMulticastProcessor.java

```diff
@@ -0,0 +1,292 @@
+        this.recipientMode = recipientMode;
+        this.filterProcessorFactorySupplier = filterProcessorFactorySupplier;
+        this.warnDroppedMessage = warnDroppedMessage;
+    }
+
+    public boolean isIgnoreInvalidEndpoints() {
+        return ignoreInvalidEndpoints;
+    }
+
+    public void setIgnoreInvalidEndpoints(boolean ignoreInvalidEndpoints) {
+        this.ignoreInvalidEndpoints = ignoreInvalidEndpoints;
+    }
+
+    protected List<Processor> createEndpointProcessors(Exchange exchange) {
+        List<String> recipientList = matchFilters(exchange).stream()
+                .map(PrioritizedFilter::getEndpoint)
+                .distinct()
+                .map(String::trim)
+                .collect(Collectors.toList());
+        if (recipientList.isEmpty()) {
+            // No matching filters, so we will use the default filter that will create a
+            // notification that there were no routing participants that matched the
+            // exchange, which results in a "dropped" message.
+            Message exchangeIn = exchange.getIn();
+            Object originalBody = exchangeIn.getBody();
+            exchangeIn.setHeader("originalBody", originalBody);
+            String endpoint = String.format(LOG_ENDPOINT, this.getClass().getCanonicalName(), getId(),
+                    warnDroppedMessage ? "WARN" : "DEBUG");
+            recipientList.add(endpoint);
+            String error = String.format(
+                    "DynamicRouter '%s': no filters matched for an exchange with id: '%s', from route: '%s'.  " +
+                                         "The 'originalBody' header contains the original message body.",
+                    getId(), exchange.getExchangeId(), exchange.getFromEndpoint());
+            exchangeIn.setBody(error, String.class);
+        }
+
+        return recipientList.stream()
+                .map(uri -> {
+                    Endpoint endpoint;
+                    try {
+                        endpoint = Optional.ofNullable(exchange.getContext().hasEndpoint(uri))
+                                .orElse(ExchangeHelper.resolveEndpoint(exchange, uri));
+                    } catch (Exception e) {
+                        if (isIgnoreInvalidEndpoints()) {
+                            LOG.debug("Endpoint uri is invalid: {}. This exception will be ignored.", uri, e);
+                            return null;
+                        } else {
+                            // failure so break out
+                            throw e;
+                        }
+                    }
+                    return endpoint;
+                })
+                .filter(Objects::nonNull)
+                .map(endpoint -> {
+                    Producer producer = producerCache.acquireProducer(endpoint);
+                    Route route = ExchangeHelper.getRoute(exchange);
+                    return wrapInErrorHandler(route, exchange, producer);
+                })
+                .toList();
+    }
```

Comment:
Re: my comment above about this being too large.

I would also move this separately and, I think, I would reconsider using the streams here as it would create iterator objects under the hood and could add pressure to the GC.

Reply:
Thanks.  I see what you mean.  Like I mentioned in the previous reply, I have simplified this a bit, so I am looking forward to your feedback.